src/main/java/io/codelex/collections/practice/phonebook/PhoneDirectory.java

```diff
@@ -1,51 +1,45 @@
 package io.codelex.collections.practice.phonebook;
 
-import java.util.Arrays;
+import java.util.TreeMap;
 
 public class PhoneDirectory {
-    private PhoneEntry[] data;
+    private final TreeMap<String,String> data;
     private int dataCount;
 
     public PhoneDirectory() {
-        data = new PhoneEntry[1];
+        data = new TreeMap<>();
         dataCount = 0;
     }
 
-    private int find(String name) {
-        for (int i = 0; i < dataCount; i++) {
-            if (data[i].name.equals(name)) {
-                return i;
-            }
+    public void putNumber(String name, String number) {
+        if (name == null || number == null) {
+            throw new IllegalArgumentException("name and number cannot be null");
         }
-        return -1;
-    }
-
-    public String getNumber(String name) {
-        int position = find(name);
-        if (position == -1) {
-            return null;
+        if (!find(name)) {
+            PhoneEntry newEntry = new PhoneEntry(name, number);
+            data.put(newEntry.getName(), newEntry.getNumber());
+            dataCount++;
+            System.out.println(name + " is added to your phone directory!");
         } else {
-            return data[position].number;
+            System.out.println(name + " is already on phone directory");
         }
     }
 
-    public void putNumber(String name, String number) {
-        if (name == null || number == null) {
-            throw new IllegalArgumentException("name and number cannot be null");
-        }
-        int i = find(name);
-        if (i >= 0) {
-            data[i].number = number;
+    private boolean find(String name) {
```

Comment:
find noteikti rada ies;aidu ka tiks meklēts kāds objekts, nevis pārbaudīts vai eksistē
un tā kā pāŗabudi tik vienu loģiku šeti arī varētu visu to likt pašā if
`if(!data.containsKey(name))`

Reply:
`if(!data.containsKey(name))` šo rindiņu IntelliJ grib īsināt uz tādu kāda jau ir. Tā vietā pārsaucu metodi, lai būtu loģiskāk!